### ISSUE TITLE
In preparation for Python3 sys.exc_clear() removal, call sys.exc_clear() on Python2 in a few key tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8496]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8496|
|Type|enhancement|
|Created|2016-06-24 08:20:57Z|
|Branch|https://github.com/twisted/twisted/tree/8496-rodrigc-sys.exc_clear|

sys.exc_clear() is gone on Python 3: https://docs.python.org/3/whatsnew/3.0.html#index-22

We should not try to call it on Python 3.

However, as we remove it, we still need to call it in a few key tests on Python2,
otherwise they fail on Python2

<details><summary>Searchable metadata</summary>

```
trac-id__8496 8496
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8496_rodrigc_sys_exc_clear 8496-rodrigc-sys.exc_clear
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1466756457213151 1466756457213151
changetime__1474198049187875 1474198049187875
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/211

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa commented|
|-|-|

Ok, high level review says this looks good. I'm sending directly to the buildbots.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa commented|
|-|-|

Ok, twistedchecker is still failing but I think that twistedchecker is losing its mind right now. I have manually verified that none of the 'new' twistedchecker errors affect the lines changed in this patch, so I think we're fine on that front.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_4"></a>|@Lukasa set owner to @rodrigc|
|-|-|

So...codecov is claiming that we're not getting complete coverage on these tests, but I think that has to be nonsense. We run the tests on Python 3 and they pass. That's...a problem. Regardless, I'm happy with this patch, we just need to resolve the idiot opinions codecov has.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset 731f444736344e578b9920476b61a17139e1bbbb
```
#!CommitTicketReference repository="" revision="731f444736344e578b9920476b61a17139e1bbbb"
Merge 8496-rodrigc-sys.exc_clear: Call sys.exc_clear() on Python 2, in few key tests 

Author: rodrigc
Reviewer: Lukasa
Fixes: #8496
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
